lib/method_source/code_helpers.rb

```diff
@@ -104,15 +104,15 @@ def extract_first_expression(lines, consume=0, &block)
     # @param [Array<String>]  lines
     # @return [String]
     def extract_last_comment(lines)
-      buffer = ""
+      buffer = +""
```

Comment:
Is this the same as `String.new("")`?

In case it is, maybe we could use it instead? It's less cryptic IMO.

Reply:
It's not quite the same. `+@` only dup the string if it's not already mutable. `String.new("")` will always create a copy.

But it doesn't make a big difference, so I don't mind changing it. But then `"".dup` might be even more readable.

Reply:
Provided `+""` is valid Ruby going back several versions (which I _think_ it is - and the CI runs should confirm that), then I personally would prefer keeping it. Yes, right now it's not common syntax, but taking a step towards normalising this in Ruby code feels smart to me - it creates a single mutable string, which is _exactly_ what's required.

`String.new("")` and `"".dup` are, as @casperisfine has noted, both duplicating strings - yes, it's a small thing, but I feel like it's not the best habit to lean into from a performance perspective.

Reply:
Ok, thanks for the clarification. This is what I found:

https://github.com/fastruby/fast-ruby?tab=readme-ov-file#string

> String#dup vs String#+ [code](https://github.com/fastruby/fast-ruby/blob/main/code/string/dup-vs-unary-plus.rb)
>
> Note that String.new is not the same as the options compared, since it is always ASCII-8BIT encoded instead of the script encoding (usually UTF-8).
>
> $ ruby -v code/string/dup-vs-unary-plus.rb
> ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-darwin17]
>
> Calculating -------------------------------------
>           String#+@      7.697M (± 1.4%) i/s -     38.634M in   5.020313s
>          String#dup      3.566M (± 1.0%) i/s -     17.860M in   5.008377s
>
> Comparison:
>           String#+@:  7697108.3 i/s
>          String#dup:  3566485.7 i/s - 2.16x  slower

Let's stick with `+""`